### PR TITLE
[KOGITO-4729] - Upgrade from Operator 1.3.0 to 1.4.0 fails with refer…

### PR DESCRIPTION
**Pull request by vaibhavjainwiz**
…ence to Kafka Topics

Jira issue : https://issues.redhat.com/browse/KOGITO-4729

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains a link to the JIRA issue
- [x] Pull Request contains a description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster
- [x] You've added a [RELEASE_NOTES.md](RELEASE_NOTES.md) entry regarding this change

### CODE REVIEW

**Comment by sutaakar** [comment]
@vaibhavjainwiz As we will support Strimzi 0.22 and newer we should switch to `v1beta2` CR version. Can you move Kafka structures to `v1beta2` folder to keep it aligned?

File: core/infrastructure/kafka/v1beta1/kafka_types.go [hunk 1]
```diff
@@ -53,20 +53,19 @@ func (kafkaMap *KafkaMap) DeepCopy() *KafkaMap {
...
-type KafkaListenerPlain struct {
+	Replicas   int32                  `json:"replicas,omitempty"`
+	Listeners  []GenericKafkaListener `json:"listeners,omitempty"`
+	Storage    KafkaStorage           `json:"storage,omitempty"`
+	Config     KafkaMap               `json:"config,omitempty"`
+	JvmOptions KafkaMap               `json:"jvmOptions,omitempty"`
+}
+
+// GenericKafkaListener ...
+type GenericKafkaListener struct {
```
Comment (ricardozanini) [inline core/infrastructure/kafka/v1beta1/kafka_types.go:64]: Why not just `KafkaListener`?

**Review by ricardozanini — COMMENTED** [review]
We need to review the types version as mentioned by @sutaakar

File: core/infrastructure/kafka/v1beta1/kafka_types.go (hunk 1 above)
Comment (vaibhavjainwiz) [reply core/infrastructure/kafka/v1beta1/kafka_types.go:64]: We could use `KafkaListener` as well. I used same name as used by Strimzi-operator developers to refer Listener.
Previously they have java class with name `KafkaListener`, but now they had mark it deprecated and create new class with name `GenericKafkaListener`

**Comment by vaibhavjainwiz** [comment]
> @vaibhavjainwiz As we will support Strimzi 0.22 and newer we should switch to `v1beta2` CR version. Can you move Kafka structures to `v1beta2` folder to keep it aligned?
> See https://strimzi.io/docs/operators/0.22.0/deploying.html#assembly-upgrade-resources-str
> Edit: Also `register.go` should be updated to reference proper version

done

**Review by spolti — APPROVED** [review]

**Review by ricardozanini — APPROVED** [review]

**Comment by sutaakar** [comment]
The code looks good, will try on OCP tomorrow.

**Review by sutaakar — APPROVED** [review]
Looks good to me, seems to work well.

**Review by radtriste — APPROVED** [review]

**Pull request merged**